test/test.py

```diff
@@ -152,52 +77,46 @@ def test_get_rate_coefficients(mechname, usr_np):
                            if sol.species_name(sp) not in
                            r.third_body.efficiencies])
             k_ct[i] *= eff
-
         print(k_ct)
         print()
         print(k_pm)
         print()
         print(np.abs((k_ct-k_pm)/k_ct))
         print(np.linalg.norm((k_ct-k_pm)/k_ct, np.inf))
         assert np.linalg.norm((k_ct-k_pm)/k_ct, np.inf) < 1e-13
-    return
 
 
 @pytest.mark.parametrize("mechname", [
     "uiuc", "sandiego", "uconn32", "gri30", "hong"
 ])
-@pytest.mark.parametrize("usr_np", numpy_list)
-def test_get_pressure(mechname, usr_np):
+@pytest.mark.parametrize("user_np", numpy_list)
+def test_get_pressure(mechname, user_np, request):
     """This function tests that pyrometheus-generated code
     computes the Cantera-predicted pressure for given density,
     temperature, and mass fractions
     """
     # Create Cantera and pyrometheus objects
-    sol = ct.Solution(f"mechs/{mechname}.yaml", "gas")
-    ptk_base_cls = pyro.codegen.python.get_thermochem_class(sol)
-    ptk = make_jax_pyro_class(ptk_base_cls, usr_np)
+    sol, ptk = pyro_init(mechname, user_np, request)
 
     # Temperature, equivalence ratio, oxidizer ratio, stoichiometry ratio
     t = 300.0
     phi = 2.0
     alpha = 0.21
     nu = 0.5
-
     # Species mass fractions
-    i_fu = ptk.species_index("H2")
-    i_ox = ptk.species_index("O2")
-    i_di = ptk.species_index("N2")
+    i_fu = ptk.get_species_index("H2")
+    i_ox = ptk.get_species_index("O2")
+    i_di = ptk.get_species_index("N2")
+    print(i_fu, i_ox, i_di)
```

Comment:
I don't know that we need to print these.

Reply:
100%, that was for debugging. Removed it in my local dev clone.